### DISCUSSION TITLE
Using the same random system in different mission elements?

### DISCUSSION THREAD

**Discussion by lobrax** [endless-sky/endless-sky#10468] [Q&A] [answered]
A general question about writing missions:

How do you use the same random (location-filtered) system for two or more different mission elements (eg, on enter, waypoint, mark, npc)? 

Specifically, I'd like to create a simple job board mission like "Go to {site}, assist {npc}, escort it to {destination}," where the random {site} comes from a location filter. I'd like the {site} name to appear in the description text and on the map so the player knows where it is before accepting. To do this, I'm trying to set a waypoint and npc to the same randomized {site}... but I can't figure out how to do that or if it's possible. 

<details> 
  <summary>For reference, I especially looked at the following missions in the game (spoiler-ish): </summary>
     <b>Remnant Rescue 4 (ie, Graveyard Distress Call)</b> job type, which is exactly how I structured mine, except that I also would like to name the system in the job description and put it on the map so the player can see it before taking the job. 
     Also, from a player perspective, the <b>FW Escort 3 mission</b> looks like what I'm going for, but is structured compeltely differently, as I'd like it to be a job board mission type that uses a randomized / location filtered system location instead of using a preset location.)
</details>

Thanks for any pointers.

**Comment by TheGiraffe3** [comment]
Looking at some missions already in the game, I don't think you can yet have an auto condition for where an NPC is. It would be a good feature request though.

**Reply by lobrax** [reply]
Agreed, i think communicating system between certain elements would open up some possibilities for building missions. I'll make a feature request if it proves not to be doable currently.

**Reply by TomGoodIdea** [reply, marked as answer]
Maybe it could be possible with #10176, via temporary conditions. But currently, I don't think it's doable.

**Reply by lobrax** [reply]
Interesting,  i hadn't considered conditions, and that issue definitely looks like it touches on my own question. I'll poke around and see if something can be cobbled together.  But I think you're right otherwise: it doesn't seem to be doable.
Thanks for the tip.

**Comment by lobrax** [comment]
Ugh, I just noticed some important {words} were hidden in the OP, so I edited them to be visible. Hopefully it makes a bit more sense now!
